modules/javafx.web/src/test/java/test/javafx/scene/web/EventListenerLeakTest.java

```diff
@@ -0,0 +1,644 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.web;
+
+import com.sun.javafx.application.PlatformImpl;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import javafx.application.Platform;
+import javafx.beans.value.ChangeListener;
+import javafx.concurrent.Worker;
+import javafx.scene.web.WebEngine;
+import javafx.scene.web.WebView;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.w3c.dom.events.Event;
+import org.w3c.dom.events.EventListener;
+import org.w3c.dom.events.EventTarget;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+// NOTE: We cannot use TestBase since we need multiple WebView instances, and
+// greater control over the lifecycle.
+public class EventListenerLeakTest {
+
+    // List of WeakReferences to EventListener objects to count which are active
+    // NOTE: this must be reset for each test
+    static List<WeakReference<?>> listenerRefs;
+
+    // Save WeakReferences to WebView objects to later check that it is released
+    // NOTE: this must be reset for each test
+    static List<WeakReference<?>> webViewRefs;
+
+    // WebView instances for testing
+    WebView webView1;
+    WebView webView2;
+
+    // List of DOM nodes for testing
+    List<EventTarget> domNodes1;
+    List<EventTarget> domNodes2;
+
+    static class MyListener implements EventListener {
+
+        private final AtomicInteger clickCount = new AtomicInteger(0);
+
+        private MyListener() {
+        }
+
+        int getClickCount() {
+            return clickCount.get();
+        }
+
+        static MyListener create() {
+            MyListener listener = new MyListener();
+            listenerRefs.add(new WeakReference<>(listener));
+            return listener;
+        }
+
+        @Override
+        public void handleEvent(Event evt) {
+            clickCount.incrementAndGet();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() throws Exception {
+        final CountDownLatch startupLatch = new CountDownLatch(1);
+
+        PlatformImpl.startup(() -> {
+            startupLatch.countDown();
+        });
+
+        assertTrue("Timeout waiting for FX runtime to start",
+                startupLatch.await(15, TimeUnit.SECONDS));
+    }
+
+    @AfterClass
+    public static void cleanupOnce() {
+        Platform.exit();
+    }
```

Comment:
I just noticed that this will cause subsequent tests to fail. Unit tests should not call `Platform.exit()` (as opposed to system tests, which should). You can remove the entire method.